### PR TITLE
Support peakSamples in query frontend response

### LINKED ISSUE #6181: Support peakSamples in query frontend response
**Is your feature request related to a problem? Please describe.**
`peakSamples` field was supported in Prometheus query response long time ago https://github.com/prometheus/prometheus/pull/10369/files#diff-9e1a49d3ecc79eae504d539895c7621d3f45eef192ddd6192d8b304ffa591864R111.

However, it is not supported yet at Cortex. https://github.com/cortexproject/cortex/blob/master/pkg/querier/tripperware/query.proto#L41

We only have `totalQueryableSamples` and `totalQueryableSamplesPerStep`.

**Describe the solution you'd like**
Modify the query frontend protobuf and add the field. Also implement the merge logic accordingly.

### PR DESCRIPTION
**Pull request by SungJin1212**
**What this PR does**:
Support peakSamples in query frontend response. The peakSamples are the maximum samples kept in memory during the query execution.
**Which issue(s) this PR fixes**:
Fixes #6181 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## master / unreleased
 
+* [FEATURE] Query Frontend: Add peakSample in query stats response. #6188
```
Comment (yeya24) [inline CHANGELOG.md:5]: This should be an enhancement not a feature.

Comment (SungJin1212) [reply CHANGELOG.md]: I've fixed it.

**Comment by yeya24** [comment]
Umm sorry @SungJin1212, https://github.com/cortexproject/cortex/pull/6180 introduced quite a lot of conflicts to this PR. Can you please rebase and address conflicts so that we can merge?

Thanks again for the great work!

**Comment by SungJin1212** [comment]
@yeya24 
I have got the rebase!

**Review by yeya24 — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**